app/controllers/reservations_controller.rb

```diff
@@ -29,16 +29,37 @@ def index # rubocop:disable CyclomaticComplexity
     filters = [:reserved, :checked_out, :overdue, :returned, :upcoming,
                :requested, :approved_requests, :denied_requests]
     filters << :missed unless AppConfig.first.res_exp_time
+    @filter = default_filter
+    @default = true
+
+    if session[:filter]
+      params[session[:filter]] = true
+      session[:filter] = nil
+    end
+
     # if the filter is defined in the params, store those reservations
     filters.each do |filter|
-      @reservations_set = @reservations_source.send(filter) if params[filter]
+      if params[filter]
+        @default = false
+        @filter = filter
+        break
+      end
     end
 
-    @default = false
-    # if no filter is defined
-    return unless @reservations_set.nil?
-    @default = true if AppConfig.first.request_text.empty?
```

Comment:
Ha I think it's a major typo from when I was cleaning up the Reservations controller for `rubocop`. I'm not sure where it came from!